editor/editor_node.cpp

```diff
@@ -6102,6 +6141,13 @@ EditorNode::EditorNode() {
 	FileAccess::set_file_close_fail_notify_callback(_file_access_close_error_notify);
 
 	waiting_for_first_scan = true;
+
+	_dimming = false;
+	_dim_time = 0.0f;
+	_dim_timer = memnew(Timer);
+	_dim_timer->set_wait_time(0.01666f);
```

Comment:
Well, I guess you can set the wait time every time before starting the timer, and using the transition time for it.